From fd4ee31896419211159d07c0d3e31f8c30196a70 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 28 Nov 2024 15:36:24 +0100
Subject: [PATCH 7/9] arch: arm: dts: cn9130-sr-som: convert pinctrl nodes
 numeric

U-Boot pinctrl driver uses numeric pin name and function values while
Linux uses strings.

Convert all pinctrl nodes used in SolidRun CN9130 SoM based platforms
accordingly.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 arch/arm/dts/cn9130-cf-base.dts     |  10 +--
 arch/arm/dts/cn9130-cf-pro.dts      |  20 +++---
 arch/arm/dts/cn9130-cf.dtsi         |  25 ++++---
 arch/arm/dts/cn9130-sr-som.dtsi     |  68 +++++++-----------
 arch/arm/dts/cn9131-cf-solidwan.dts | 106 ++++++++++++++--------------
 5 files changed, 103 insertions(+), 126 deletions(-)

diff --git a/arch/arm/dts/cn9130-cf-base.dts b/arch/arm/dts/cn9130-cf-base.dts
index 0d7d2ddc951..7b123b2d6d8 100644
--- a/arch/arm/dts/cn9130-cf-base.dts
+++ b/arch/arm/dts/cn9130-cf-base.dts
@@ -140,18 +140,18 @@
 	};
 };
 
-&cp0_pinctrl {
+&cp0_pinctl {
 	pinctrl-0 = <&sim_select_pins>;
 	pintrl-names = "default";
 
 	rear_button_pins: cp0-rear-button-pins {
-		marvell,pins = "mpp31";
-		marvell,function = "gpio";
+		marvell,pins = <31>;
+		marvell,function = <0>;
 	};
 
 	sim_select_pins: cp0-sim-select-pins {
-		marvell,pins = "mpp27";
-		marvell,function = "gpio";
+		marvell,pins = <27>;
+		marvell,function = <0>;
 	};
 };
 
diff --git a/arch/arm/dts/cn9130-cf-pro.dts b/arch/arm/dts/cn9130-cf-pro.dts
index 9c5ffafd331..5874fed2c2e 100644
--- a/arch/arm/dts/cn9130-cf-pro.dts
+++ b/arch/arm/dts/cn9130-cf-pro.dts
@@ -325,31 +325,31 @@
 	status = "okay";
 };
 
-&cp0_pinctrl {
+&cp0_pinctl {
 	dsa_clk_pins: cp0-dsa-clk-pins {
-		marvell,pins = "mpp40";
-		marvell,function = "synce1";
+		marvell,pins = <40>;
+		marvell,function = <2>;
 	};
 
 	dsa_pins: cp0-dsa-pins {
-		marvell,pins = "mpp27", "mpp29";
-		marvell,function = "gpio";
+		marvell,pins = <27 29>;
+		marvell,function = <0>;
 	};
 
 	rear_button_pins: cp0-rear-button-pins {
-		marvell,pins = "mpp32";
-		marvell,function = "gpio";
+		marvell,pins = <32>;
+		marvell,function = <0>;
 	};
 
 	cp0_spi1_cs1_pins: cp0-spi1-cs1-pins {
-		marvell,pins = "mpp12";
-		marvell,function = "spi1";
+		marvell,pins = <12>;
+		marvell,function = <3>;
 	};
 };
 
 &cp0_spi1 {
 	/* add pin for chip-select 1 on mikrobus */
-	pinctrl-0 = <&cp0_spi1_pins &cp0_spi1_cs1_pins>;
+	pinctrl-0 = <&cp0_spi0_pins &cp0_spi1_cs1_pins>;
 };
 
 /* USB-2.0 Host on Type-A connector */
diff --git a/arch/arm/dts/cn9130-cf.dtsi b/arch/arm/dts/cn9130-cf.dtsi
index 7faa8ee57a6..e03cf90aeee 100644
--- a/arch/arm/dts/cn9130-cf.dtsi
+++ b/arch/arm/dts/cn9130-cf.dtsi
@@ -144,26 +144,25 @@
 	status = "okay";
 };
 
-&cp0_pinctrl {
-	cp0_mmc0_pins: cp0-mmc0-pins {
-		marvell,pins = "mpp43", "mpp56", "mpp57", "mpp58",
-			       "mpp59", "mpp60", "mpp61";
-		marvell,function = "sdio";
+&cp0_pinctl {
+	cp0_mmc0_cd_pins: cp0-mmc0-cd-pins {
+		marvell,pins = <43>;
+		marvell,function = <1>;
 	};
 
 	mikro_spi_pins: cp0-spi1-cs1-pins {
-		marvell,pins = "mpp12";
-		marvell,function = "spi1";
+		marvell,pins = <12>;
+		marvell,function = <3>;
 	};
 
 	mikro_uart_pins: cp0-uart-pins {
-		marvell,pins = "mpp2", "mpp3";
-		marvell,function = "uart1";
+		marvell,pins = <2 3>;
+		marvell,function = <8>;
 	};
 
 	expander0_pins: cp0-expander0-pins {
-		marvell,pins = "mpp4";
-		marvell,function = "gpio";
+		marvell,pins = <4>;
+		marvell,function = <0>;
 	};
 };
 
@@ -177,7 +176,7 @@
 
 /* microSD */
 &cp0_sdhci0 {
-	pinctrl-0 = <&cp0_mmc0_pins>;
+	pinctrl-0 = <&cp0_sdhci_pins>, <&cp0_mmc0_cd_pins>;
 	pinctrl-names = "default";
 	bus-width = <4>;
 	no-1-8-v;
@@ -186,7 +185,7 @@
 
 &cp0_spi1 {
 	/* CS1 for mikrobus */
-	pinctrl-0 = <&cp0_spi1_pins &mikro_spi_pins>;
+	pinctrl-0 = <&cp0_spi0_pins &mikro_spi_pins>;
 };
 
 /*
diff --git a/arch/arm/dts/cn9130-sr-som.dtsi b/arch/arm/dts/cn9130-sr-som.dtsi
index f688e4cc229..b6a552f6e8a 100644
--- a/arch/arm/dts/cn9130-sr-som.dtsi
+++ b/arch/arm/dts/cn9130-sr-som.dtsi
@@ -45,28 +45,24 @@
 
 &ap_pinctl {
 	ap_mmc0_pins: ap-mmc0-pins {
-		marvell,pins = "mpp0", "mpp1", "mpp2", "mpp3", "mpp4", "mpp5",
-					   "mpp6", "mpp7", "mpp8", "mpp9", "mpp10", "mpp12";
-		marvell,function = "sdio";
-		/*
-		 * mpp12 is emmc reset, function should be sdio (hw_rst),
-		 * but pinctrl-mvebu does not support this.
-		 *
-		 * From pinctrl-mvebu.h:
-		 * "The name will be used to switch to this setting in DT description, e.g.
-		 * marvell,function = "uart2". subname is only for debugging purposes."
-		 */
+		marvell,pins = <0 1 2 3 4 5 6 7 8 9 10>;
+		marvell,function = <1>;
 	};
 
-	ap_uart0_pins: ap-uart0-pins {
-		marvell,pins = "mpp11", "mpp19";
-		marvell,function = "uart0";
+	ap_mmc0_rst_pins: ap-mmc0-rst-pins {
+		marvell,pins = <12>;
+		marvell,function = <2>;
+	};
+
+	uart0_pins: uart0-pins {
+		marvell,pins = <11 19>;
+		marvell,function = <3>;
 	};
 };
 
 &ap_sdhci0 {
 	bus-width = <8>;
-	pinctrl-0 = <&ap_mmc0_pins>;
+	pinctrl-0 = <&ap_mmc0_pins>, <&ap_mmc0_rst_pins>;
 	pinctrl-names = "default";
 	vqmmc-supply = <&v_1_8>;
 	status = "okay";
@@ -79,7 +75,7 @@
 /* for assembly with phy */
 &cp0_eth2 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&cp0_eth2_pins>;
+	pinctrl-0 = <&cp0_ge2_rgmii_pins>;
 	phy-mode = "rgmii-id";
 	phy = <&cp0_eth2_phy>;
 	status = "okay";
@@ -100,6 +96,7 @@
 
 &cp0_mdio {
 	pinctrl-0 = <&cp0_mdio_pins>;
+	pinctrl-names = "default";
 	status = "okay";
 
 	/* assembly option */
@@ -111,7 +108,7 @@
 &cp0_spi1 {
 	status = "okay";
 	pinctrl-names = "default";
-	pinctrl-0 = <&cp0_spi1_pins>;
+	pinctrl-0 = <&cp0_spi0_pins>;
 	/* max speed limited by a mux */
 	spi-max-frequency = <1800000000>;
 
@@ -123,38 +120,21 @@
 	};
 };
 
-&cp0_syscon0 {
-	cp0_pinctrl: pinctrl {
-		compatible = "marvell,cp115-standalone-pinctrl";
-
-		cp0_eth2_pins: cp0-ge2-rgmii-pins {
-			marvell,pins = "mpp44", "mpp45", "mpp46", "mpp47",
-				       "mpp48", "mpp49", "mpp50", "mpp51",
-				       "mpp52", "mpp53", "mpp54", "mpp55";
-			/* docs call it "ge2", but cp110-pinctrl "ge1" */
-			marvell,function = "ge1";
-		};
-
-		cp0_mdio_pins: cp0-mdio-pins {
-			marvell,pins = "mpp40", "mpp41";
-			marvell,function = "ge";
-		};
-
-		cp0_spi1_pins: cp0-spi1-pins {
-			marvell,pins = "mpp13", "mpp14", "mpp15", "mpp16";
-			marvell,function = "spi1";
-		};
-
-		cp0_reg_vhv_pins: cp0-reg-vhv-pins {
-			marvell,pins = "mpp41";
-			marvell,function = "gpio";
-		};
+&cp0_pinctl {
+	cp0_mdio_pins: cp0-mdio-pins {
+		marvell,pins = <0 1>;
+		marvell,function = <10>;
+	};
+
+	cp0_reg_vhv_pins: cp0-reg-vhv-pins {
+		marvell,pins = <41>;
+		marvell,function = <0>;
 	};
 };
 
 /* AP default console */
 &uart0 {
-	pinctrl-0 = <&ap_uart0_pins>;
+	pinctrl-0 = <&uart0_pins>;
 	pinctrl-names = "default";
 	status = "okay";
 };
diff --git a/arch/arm/dts/cn9131-cf-solidwan.dts b/arch/arm/dts/cn9131-cf-solidwan.dts
index 19e0cfe7739..89bb81c1b44 100644
--- a/arch/arm/dts/cn9131-cf-solidwan.dts
+++ b/arch/arm/dts/cn9131-cf-solidwan.dts
@@ -336,55 +336,54 @@
 	status = "okay";
 };
 
-&cp0_pinctrl {
+&cp0_pinctl {
 	pinctrl-0 = <&cp0_m2_0_shutdown_pins &cp0_mpcie_rfkill_pins>;
 	pinctrl-names = "default";
 
 	cp0_led_pins: cp0-led-pins {
-		marvell,pins = "mpp4", "mpp7";
-		marvell,function = "gpio";
+		marvell,pins = <4 7>;
+		marvell,function = <0>;
 	};
 
 	cp0_m2_0_shutdown_pins: cp0-m2-0-shutdown-pins {
-		marvell,pins = "mpp8";
-		marvell,function = "gpio";
+		marvell,pins = <8>;
+		marvell,function = <0>;
 	};
 
-	cp0_mmc0_pins: cp0-mmc0-pins {
-		marvell,pins = "mpp43", "mpp56", "mpp57", "mpp58",
-			       "mpp59", "mpp60", "mpp61";
-		marvell,function = "sdio";
+	cp0_mmc0_cd_pins: cp0-mmc0-cd-pins {
+		marvell,pins = <43>;
+		marvell,function = <1>;
 	};
 
 	cp0_mpcie_rfkill_pins: cp0-mpcie-rfkill-pins {
-		marvell,pins = "mpp6";
-		marvell,function = "gpio";
+		marvell,pins = <6>;
+		marvell,function = <0>;
 	};
 
 	cp0_reg_usb_a_vbus0_pins: cp0-reg-usb-a-vbus0-pins {
-		marvell,pins = "mpp27";
-		marvell,function = "gpio";
+		marvell,pins = <27>;
+		marvell,function = <0>;
 	};
 
 	cp0_reg_usb_a_vbus1_pins: cp0-reg-usb-a-vbus1-pins {
-		marvell,pins = "mpp28";
-		marvell,function = "gpio";
+		marvell,pins = <28>;
+		marvell,function = <0>;
 	};
 
 	cp0_sfp0_pins: cp0-sfp0-pins {
-		marvell,pins = "mpp31", "mpp32", "mpp33", "mpp34";
-		marvell,function = "gpio";
+		marvell,pins = <31 32 33 34>;
+		marvell,function = <0>;
 	};
 
 	cp0_spi1_cs1_pins: cp0-spi1-cs1-pins {
-		marvell,pins = "mpp12";
-		marvell,function = "spi1";
+		marvell,pins = <12>;
+		marvell,function = <3>;
 	};
 };
 
 /* microSD */
 &cp0_sdhci0 {
-	pinctrl-0 = <&cp0_mmc0_pins>;
+	pinctrl-0 = <&cp0_sdhci_pins>, <&cp0_mmc0_cd_pins>;
 	pinctrl-names = "default";
 	bus-width = <4>;
 	no-1-8-v;
@@ -393,7 +392,7 @@
 
 &cp0_spi1 {
 	/* add pin for chip-select 1 */
-	pinctrl-0 = <&cp0_spi1_pins &cp0_spi1_cs1_pins>;
+	pinctrl-0 = <&cp0_spi0_pins &cp0_spi1_cs1_pins>;
 
 	flash@1 {
 		compatible = "jedec,spi-nor";
@@ -604,46 +603,45 @@
 	/delete-node/ sata-port@1;
 };
 
-&cp1_syscon0 {
-	cp1_pinctrl: pinctrl {
-		compatible = "marvell,cp115-standalone-pinctrl";
-		pinctrl-0 = <&cp1_m2_1_shutdown_pins &cp1_m2_2_shutdown_pins>;
-		pinctrl-names = "default";
+&cp1_pinctl {
+	compatible = "marvell,mvebu-pinctrl", "marvell,armada-8k-cpm-pinctrl";
+	bank-name ="cp1-110";
+	pinctrl-0 = <&cp1_m2_1_shutdown_pins &cp1_m2_2_shutdown_pins>;
+	pinctrl-names = "default";
 
-		cp1_i2c1_pins: cp0-i2c1-pins {
-			marvell,pins = "mpp35", "mpp36";
-			marvell,function = "i2c1";
-		};
+	cp1_i2c1_pins: cp0-i2c1-pins {
+		marvell,pins = <35 36>;
+		marvell,function = <2>;
+	};
 
-		cp1_led_pins: cp1-led-pins {
-			marvell,pins = "mpp54", "mpp55";
-			marvell,function = "gpio";
-		};
+	cp1_led_pins: cp1-led-pins {
+		marvell,pins = <54 55>;
+		marvell,function = <0>;
+	};
 
-		cp1_m2_1_shutdown_pins: cp1-m2-1-shutdown-pins {
-			marvell,pins = "mpp29";
-			marvell,function = "gpio";
-		};
+	cp1_m2_1_shutdown_pins: cp1-m2-1-shutdown-pins {
+		marvell,pins = <29>;
+		marvell,function = <0>;
+	};
 
-		cp1_m2_2_shutdown_pins: cp1-m2-2-shutdown-pins {
-			marvell,pins = "mpp30";
-			marvell,function = "gpio";
-		};
+	cp1_m2_2_shutdown_pins: cp1-m2-2-shutdown-pins {
+		marvell,pins = <30>;
+		marvell,function = <0>;
+	};
 
-		cp1_mdio_pins: cp1-mdio-pins {
-			marvell,pins = "mpp37", "mpp38";
-			marvell,function = "ge";
-		};
+	cp1_mdio_pins: cp1-mdio-pins {
+		marvell,pins = <37 38>;
+		marvell,function = <7>;
+	};
 
-		cp1_rtc_pins: cp1-rtc-pins {
-			marvell,pins = "mpp12", "mpp13";
-			marvell,function = "gpio";
-		};
+	cp1_rtc_pins: cp1-rtc-pins {
+		marvell,pins = <12 13>;
+		marvell,function = <0>;
+	};
 
-		cp1_sfp1_pins: cp1-sfp1-pins {
-			marvell,pins = "mpp33", "mpp34", "mpp49", "mpp50";
-			marvell,function = "gpio";
-		};
+	cp1_sfp1_pins: cp1-sfp1-pins {
+		marvell,pins = <33 34 49 50>;
+		marvell,function = <0>;
 	};
 };
 
-- 
2.43.0

